### ISSUE TITLE
[Bug]: Notification clicks not triggered on Windows when App User Model ID matches a valid app

### ISSUE THREAD

**Issue by sergiou87** [electron/electron#32585]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

16.0.7

### What operating system are you using?

Windows

### Operating System Version

Windows 10 Pro 19044

### What arch are you using?

x64

### Expected Behavior

`click` events are notified to the app when the App User Model ID of the process matches the one of the app's shortcut.

### Actual Behavior

`click` events are **not** notified to the app when the App User Model ID of the process matches the one of the app's shortcut.

Other events like `show` or `close` are. Notifications are displayed as expected too.

### Testcase Gist URL

https://gist.github.com/sergiou87/ac82e95f428fc8438a7d794ad28195b6

### Additional Information

If I change the process App User Model ID to something that doesn't match any app's shortcuts or I remove the app's shortcuts, the `click` events are reported back to the app. However, the notifications in that scenario don't show the app's title and icon, so they look a bit sketchy.

I've tried this also on the latest Windows 11 for ARM with the same result.

Another thing I tried was using [node-notifier](https://github.com/mikaelbr/node-notifier/) (or, more specifically, [snoretoast](https://github.com/KDE/snoretoast)). With it I still had the very same issue, but it allowed me to test something else: running `snoretoast` from the command line with the `-install` parameter, I could create a new shortcut with the App User Model ID (and replace the one generated by Squirrel installer).
`snoretoast-x64.exe -install "Electron Fiddle" "C:/.../electron-fiddle.exe" "com.squirrel.electron-fiddle.electron-fiddle"`

With the new shortcut, notifications from `node-notifier` / `snoretoast` report `click` back to the app, but Electron's notifications still don't!!

Summary about when I get `click` events in the app:
| | Shorctut from Squirrel installer | Shorcut created with `snoretoast` |
|-|-|-|
| Electron `Notification` module | ❌ | ❌ |
| HTML5 Notifications API | ❌ | ❌ |
| `node-notifier` / `snoretoast` notifications | ❌ | ✅ |

This makes me thing there is something broken in Squirrel's installer when it creates the shorcut, and something broken in Electron that even with an apparently good shortcut can't get `click` events.

**Comment by sergiou87** [comment]
About that table, it seems the key was that the `toastActivatorClsid` of the "good" shortcut matches the one used by snoretoast here:
- https://github.com/KDE/snoretoast/blob/2fae686f5826ac4c407c772db56873f538a384ba/CMakeLists.txt#L5
- https://github.com/KDE/snoretoast/blob/31947076311ab13efb0758288cc20dfbfbf2c93d/src/snoretoastactioncenterintegration.h#L45

I haven't found anything similar in Electron which I guess is the problem.

I tried just removing the `toastActivatorClsid` that Squirrel generates from the shortcuts and it works!! I'm getting clicks notified to the app. They still don't work when notifications are clicked from the Windows Action Center, though. For that I think Electron might need a COM class like that last snoretoast link I shared.
